<a href="https://colab.research.google.com/github/gyujin0450/RPRG/blob/master/R11_%ED%99%95%EB%A5%A0%EC%98%88%EC%A0%9C.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 추론 통계 개요
기술통계는 수집한 자료가 어떤 분포와 특성을 가겼는지 분석
추론통계는 통계자/분석자가 세운 통계적 가설을
확률기반으로 채택여부를 판단하는 통계 기법

# 추론통계에서 중요한 개념
모집단, 표본, 표본공간, 사건, 확률


## 확률
1) 동전 1개를 10회 던져보고, 뒤면의 수와 앞면의 수룰 출력

In [22]:
# sample(표본공간, 횟수, 추출방법)
# 복원추출 : replace = T 
# 비복원추출 : replace = F 
r <- sample(c("앞","뒤"), 10, replace = T)
table(r)

r
뒤 앞 
 2  8 

2) 로또 번호 10개 생성

In [69]:
n <- c(1:10)
sink("result1.txt", append = T)
for(i in n) {
  r <- sample(c(1:45), 6, replace = F)
  cat(i, ":", r, "\n")
}
sink()  

1 : 41 44 34 43 38 29 
2 : 35 34 10 43 7 39 
3 : 44 32 39 11 38 18 
4 : 34 19 10 40 30 36 
5 : 37 41 33 40 28 15 
6 : 36 42 32 3 29 22 
7 : 38 41 33 6 16 20 
8 : 26 1 14 8 37 28 
9 : 27 32 25 23 3 7 
10 : 27 9 42 25 31 26 


3) 주사위 1개를 60,000회 던져보고, 각 눈의 나온 수룰 출력

In [36]:
r <- sample(c(1:6), 60000, replace = T)
table(r)

r
    1     2     3     4     5     6 
10110  9850  9948 10056  9882 10154 

4) 1 ~ 45 사이 임의의수 6개를 비복원출방식으로 뽑아보세요

In [38]:
sample(c(1:45), 6, replace = F)

[1] 29  5 32 39  6 12

5) 1 ~ 45 사이 임의의수 6개를 복원출방식으로 뽑아보세요

In [41]:
sample(c(1:45), 6, replace = T)

[1] 24 13 37 38 18  8

# 주관적 확률 - 나이브베이즈
https://nittaku.tistory.com/283

사전확률 : 관측자가 관측하기 전에 알고 있는 확률
- 확률 시행 전에 이미 가지고 있는 지식을 통해 부여한 확률 

사후확률 : 사건 발생후에 어떤 원인으로 부터 일어난 것이라고 생각되어지는 확률
- 추가된 정보로 부터 사전정보를 새롭게 수정한 확률 (수정 확률) 
- 베이즈 정리로 부터 구할 수 있음


# 베이즈 이론 활용
- 메일제목으로 메일종류 분석 - 스팸 분류
- 개인나이,성별,직업등의 데이터를 토대로 영화장르 추천
- 나이,성별,검진정보를 토대로 심장병 여부 판별

# 곱셈법칙

상호종속적 두사건에 대한 곱셈법칙은
다른 사건에 대한 조건부 확률과 주어진 사건의 확률의 곱을 의미
- P(A n B ) = P(A)P(B|A)

이때 곱셈법칙의 동등석을 이용해서 우변을 다음과 같이 변경
- P(A n B ) = P(A)P(B|A) = P(A|B)P(B)

이것을 이용해서 베이즈 정리를 유도할 수 있음
- P(A)P(B|A) = P(A|B)P(B)
- P(A|B) = P(A)P(B|A) / P(B) = P(B)P(A|B) / P(A)


베이즈 확률 vs 경험적 확률
- 동전 100회 던졌을 때 앞면이 나올 확율은?
- 경험적 확률에서는 '일어난 횟수/전체시도횟수'로 계산

만일, A라는 도시에 수지가 태어난 경우
이 아이가 노벨상을 받을 확률은?
- 이것을 경험적 확률로 계산하려면? 거의 불가능
- 이러한 상황에서는 나이브베이즈 확률을 이용함
- 일어나지 않은 일에 대한 확률을 불확실성이라는 개념으로 말함


예) 베이즈 정리 예제 (1)
- 어떤 사람이 호흡하기 어려울 정도의 가슴 통증 유발
- 폐암에 걸릴 확률은 1% 정도임
- 병원 방문 후 정확도 90% 검사 실행 결과가 양성일때, 과연 폐암일까?



양성일때 폐암일 확률 
- P(폐암|양성) = P(폐암)P(양성|폐암) / P(양성)
- 검사했을 때 폐암의 확률 : 90% => P(양성|폐암)
- 폐암에 걸리 확률 : 1% => P(폐암)
- 따라서, P(폐암)P(양성|폐암) = 0.01 * 0.9 

중요! - P(양성) => 양성일 조건
- 폐암이고 검사후 실제로 양성인 경우
- 폐암이 아닌데, 검사후 양성인 경우

** 전체 확률의 정리 **
- P(양성) = P(양성|폐암)P(폐암) + P(양성|1-폐암)P(1-폐암) 
          = (0.9 * 0.01) + (0.1 * 0.99)
          = 0.108

따라서, P(폐암|양성) = (0.9 * 0.01) / 0.108 = 0.08333 ==> 8.3%

결론 : 별거 아님



예제] 동호회 회원수가 100명 일 때 여성은 40명, 남성은 60명이다.
- 이중 기혼인 여성은 16명, 기혼인 남성은 30명일때
- 임의로 뽑은 회원이 기혼이라 할 때 여성일 확율은?

P(여성|기혼) = P(여성)P(기혼|여성) / P(기혼)
- P(여성) : 0.4
- P(기혼|여성) ~ 여성 중 기혼 : 0.4
- P(기혼) ~ 여성이면서 기혼 + 남성이면서 기혼
          = P(기혼|여성)P(여성) + P(기혼|1-여성)P(1-여성)
          = 0.4 * 0.4 + (30/60) * 0.6
          = 0.46
- P(여성|기혼) = 0.4 * 0.4 / 0.46 
          = 0.348           

어떤 나라에서 시민의 39%가 여성이라고 가정해 보자.
우리는 또한 여성의 22%와 남성의 14%가 직업이 없다는 것을 알고 있다.
그렇다면 이 나라의 인구 중에서 무작위로 뽑은 사람이 실업자 P(U)일 확률은 얼마나 될까?
https://github.com/gyujin0450/Rstudy/blob/master/R11_%ED%99%95%EB%A5%A0%EC%98%88%EC%A0%9C.ipynb

예) 베이즈 정리 예제 (2)
- 시간이 지나 다시 병원을 찾음
- 정확도 99% 검사 실행
- 결과가 양성
- 의사 : 폐암에 걸리 확률이 50%로 증가
- 과연 폐암일까?
